ydb/core/cms/console/console_handshake.cpp

```diff
@@ -0,0 +1,104 @@
+
+namespace NKikimr::NConsole {
+
+void TConfigsManager::Handle(TEvBlobStorage::TEvControllerProposeConfigRequest::TPtr &ev, const TActorContext &ctx) {
+    const auto& record = ev->Get()->Record;
+    const auto& proposedConfigHash = record.GetConfigHash();
+    const auto& proposedConfigVersion = record.GetConfigVersion();
+
+    auto currentConfigHash = std::hash<std::string>{}(YamlConfig);
+    if (Self.CurrentSenderId != ev->Sender) {
+        auto endSessionEvent = std::make_unique<IEventHandle>(Self.SelfId(), Self.CurrentSenderId, new TEvConsole::TEvCloseSession());
+        if (Self.CurrentInterconnectSession) {
+            endSessionEvent->Rewrite(TEvInterconnect::EvForward, Self.CurrentInterconnectSession);
+            ctx.Send(endSessionEvent.release());
+        }
+        else {
+            return;
+        }
+    }
+    Self.CurrentSenderId = ev->Sender;
+    Self.CurrentPipeServerId = ev->Recipient;
+    Self.CurrentInterconnectSession = ev->InterconnectSession;
+    auto response = std::make_unique<TEvBlobStorage::TEvControllerProposeConfigResponse>();
+    auto& responseRecord = response->Record;
+
+    if (YamlVersion > record.GetConfigVersion()) {
+        responseRecord.SetStatus(NKikimrBlobStorage::TEvControllerProposeConfigResponse::UnexpectedConfig);
+        responseRecord.SetProposedConfigVersion(proposedConfigVersion);
+        responseRecord.SetConsoleConfigVersion(YamlVersion);
+        SendInReply(ev->Sender, Self.CurrentInterconnectSession, std::move(response));
+        LOG_ALERT_S(ctx, NKikimrServices::CMS, "Unexpected proposed config.");
+        return;
+    }
+    if (YamlVersion == record.GetConfigVersion()) {
+        responseRecord.SetStatus(NKikimrBlobStorage::TEvControllerProposeConfigResponse::CommitIsNotNeeded);
+        SendInReply(ev->Sender, Self.CurrentInterconnectSession, std::move(response));
+        return;
+    }
+    if (proposedConfigHash != currentConfigHash) {
+        responseRecord.SetStatus(NKikimrBlobStorage::TEvControllerProposeConfigResponse::HashMismatch);
+        responseRecord.SetProposedConfigHash(proposedConfigHash);
+        responseRecord.SetConsoleConfigHash(currentConfigHash);
+        SendInReply(ev->Sender, Self.CurrentInterconnectSession, std::move(response));
+        LOG_ALERT_S(ctx, NKikimrServices::CMS, "Config hash mismatch.");
+        return;
+    }
+    responseRecord.SetStatus(NKikimrBlobStorage::TEvControllerProposeConfigResponse::CommitIsNeeded);
+    SendInReply(ev->Sender, Self.CurrentInterconnectSession, std::move(response));
+}
+
+void TConfigsManager::Handle(TEvBlobStorage::TEvControllerConsoleCommitRequest::TPtr &ev, const TActorContext &ctx) {
+    auto response = std::make_unique<TEvBlobStorage::TEvControllerConsoleCommitResponse>();
+    auto& record = response->Record;
+    const auto& yamlConfig = ev->Get()->Record.GetYAML();
+
+    if (Self.CurrentSenderId != ev->Sender || Self.CurrentPipeServerId != ev->Recipient) {
```

Comment:
it would be more clear if you'd use the following construct:
if (CurrentSenderId != ExpectedSenderId) {
...
} else if (CurrentPipeId != ExpectedPipeId) {
...
}